### ISSUE TITLE
Imgur download fails (on Windows) if image has ?1 in url

### ISSUE THREAD

**Issue by ikoutone** [RipMeApp/ripme#418]
* Ripme version: 1.17.14
* Java version:  java version "1.8.0_161"
* Operating system: Microsoft Windows [Version 10.0.16299.192]
* Exact URL you were trying to rip when the problem occurred: https://imgur.com/gallery/UUkPK
* Please include any additional information about how to reproduce the problem:

## Expected Behavior
All images are downloaded

## Actual Behavior
Detail the actual (incorrect) behavior here. You can post log snippets or attach log files to your issue report.

Ripme enters an endless loop in the log tab which keeps repeating the image it can't download
Downloading http://i.imgur.com/HjoXJAd.jpg?1

**Comment by cyian-1756** [comment]
I can't repo this with ripme 1.7.14 with "1.8.0_151"

**Comment by ddbreight** [comment]
Same issue with 1.7.16 on this gallery https://imgur.com/a/hgUyd

error
https://imgur.com/a/1AicE

**Comment by cyian-1756** [comment]
@ddbreight I still can't repo this. Can you post ripmes log file?

**Comment by ikoutone** [comment]
Here is my log from the issue.

[log.txt](https://github.com/RipMeApp/ripme/files/1713730/log.txt)

**Comment by cyian-1756** [comment]
```
2018-02-10 22:29:03,547 ERROR utils.Utils .removeCWD() Exception: 
java.io.IOException: Invalid argument
	at java.io.WinNTFileSystem.canonicalize0(Native Method)
	at java.io.WinNTFileSystem.canonicalize(Unknown Source)
	at java.io.File.getCanonicalPath(Unknown Source)
	at com.rarchives.ripme.utils.Utils.removeCWD(Utils.java:220)
	at com.rarchives.ripme.ripper.DownloadFileThread.<init>(DownloadFileThread.java:46)
	at com.rarchives.ripme.ripper.AlbumRipper.addURLToDownload(AlbumRipper.java:84)
	at com.rarchives.ripme.ripper.AlbumRipper.addURLToDownload(AlbumRipper.java:98)
	at com.rarchives.ripme.ripper.rippers.ImgurRipper.ripAlbum(ImgurRipper.java:216)
	at com.rarchives.ripme.ripper.rippers.ImgurRipper.ripAlbum(ImgurRipper.java:187)
	at com.rarchives.ripme.ripper.rippers.ImgurRipper.rip(ImgurRipper.java:164)
	at com.rarchives.ripme.ripper.AbstractRipper.run(AbstractRipper.java:473)
	at java.lang.Thread.run(Unknown Source)
2018-02-10 22:29:04,101 ERROR ripper.DownloadFileThread .run() [!] Exception while downloading file: http://i.imgur.com/HjoXJAd.jpg?1 - D:\Zipit\rips\imgur_UUkPK_goTHT\016_HjoXJAd.jpg?1 (The filename, directory name, or volume label syntax is incorrect)
```

Looks like this is a file system issue. What file system are you using on D: (NTFS, FAT, XFAT ect)? Does that mount point exist? Can you create a file named "HjoXJAd.jpg?1" there?

**Comment by ikoutone** [comment]
NTFS and yes, the path exists and all other pictures from the imgur album download fine. 

File names in Windows can't contain a ?, so that must be the problem.

**Comment by cyian-1756** [comment]
> File names in Windows can't contain a ?

Well damn. I'll add ? to the char blacklist and push out an update, it should be done sometime tomorrow

**Comment by cyian-1756** [comment]
This should be fixed in the 1.7.18 update

**Comment by ikoutone** [comment]
Unfortunately it broke the imgur download completely for me. Trying to download the album https://imgur.com/gallery/UUkPK gets me this in log window.

Downloading http://imgur.com/a/UUkPK
Downloading http://i.imgur.com/371zVjb.jpg
Downloading http://i.imgur.com/uMyQn2p.jpg
Downloading http://i.imgur.com/NYZY5AE.jpg
Downloading http://i.imgur.com/cZOeAOw.jpg
Downloading http://i.imgur.com/GoFeU68.jpg
Downloaded .\D
Downloading http://i.imgur.com/kshdHYE.jpg
Downloaded .\D
Downloading http://i.imgur.com/gImTyms.jpg
Downloaded .\D
Downloading http://i.imgur.com/kBXwOwu.jpg
Downloaded .\D
Downloading http://i.imgur.com/vXluNy2.jpg
Downloaded .\D
Downloading http://i.imgur.com/9ahF63H.jpg
Downloaded .\D
Downloading http://i.imgur.com/0Q9BlJ4.jpg
Downloaded .\D
Downloading http://i.imgur.com/nzrKxM9.jpg
Downloaded .\D
Downloading http://i.imgur.com/wGmtCfB.jpg
Downloaded .\D
Downloading http://i.imgur.com/tZ1CqB7.png
Downloaded .\D
Downloading http://i.imgur.com/nGsCZkH.jpg
Downloaded .\D
Downloading http://i.imgur.com/HjoXJAd.jpg?1
Downloaded .\D
Downloading http://i.imgur.com/LURGoUE.jpg
Downloaded .\D
Downloaded .\D
Downloaded .\D
Downloaded .\D
Downloaded .\D
Rip complete, saved to D:\Zipit\rips\imgur_UUkPK_goTHT

Nothing except log file is saved to the folder, and file called D is created in ripme folder.

[log.txt](https://github.com/RipMeApp/ripme/files/1714457/log.txt)

**Comment by cyian-1756** [comment]
Oh bloody hell.

**Comment by cyian-1756** [comment]
I've fixed this again (Hopefully for real this time) in 1.7.19

**Comment by cyian-1756** [comment]
@ikoutone Can you confirm this fixes the issue?

**Comment by ikoutone** [comment]
Yes, it's fixed in the 1.7.19. Thank you.
